### PR TITLE
Add 'biotite' package to conda-forge

### PR DESCRIPTION
**Pull request by padix-key**

### CODE REVIEW

**Comment by hajapy** [comment]
Update your branch with upstream/master and this should go away.

**Comment by dougalsutherland** [comment]
A rebuild should be enough; I hit restart.

**Comment by dougalsutherland** [comment]
Hi @padix-key, thanks for the submission! I just made a few small changes:

- Because conda needs to build a separate package for each python version, the plain `python >=3.6` syntax doesn't work for python itself. Instead you want to leave `python` plain and add a `skip` statement like I did here. (This is what caused the Windows crash before, though it would have been broken on every platform anyway.)
- You generally want to install with the oldest working version of numpy; `numpy >=1.13` would pick the latest one, and so the builds would unnecessarily not be compatible with slightly older `numpy`s.
- I switched to installing with `pip`, which is the generally preferred way in conda-forge.

**Comment by padix-key** [comment]
Thank you for your help. I modified the Python version selector, since the package is not compatible with Python 3.5. I think the pull request is now ready to be merged.

**Comment by padix-key** [comment]
@conda-forge/core

**Comment by padix-key** [comment]
Currently the following error occurs in the continuous integration build on all platforms:
```
conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
  - numpy=1.13 -> python[version='>=2.7,<2.8.0a0']
  - python=3.7
```

This error does not occur in my local builds. Is this a problem in the continuous integration or in the recipe?

**Comment by padix-key** [comment]
The `win_64 win` build gives a cryptic Python error traceback:
```
Generating script.
##[command]"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\1a90c4b6-49f3-4bc2-bcbb-09a03ae993d2.cmd""
Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
INFO:conda_build.variants:Adding in variants from internal_defaults
Building biotite,sphinxcontrib-fulltoc with conda-forge/label/main
rendering D:\a\1\s\recipes\biotite for win-win-64
rendering D:\a\1\s\recipes\sphinxcontrib-fulltoc for win-win-64
Computed that there are 3 distributions to build from 2 recipes
Resolved dependencies, will be built in the following order:
    biotite-0.12.0-c_compiler_vs2015-python_3.6-target_win-64-on-win-win-64
    biotite-0.12.0-c_compiler_vs2015-python_3.7-target_win-64-on-win-win-64
    sphinxcontrib-fulltoc-1.2.0--target_win-64-on-win-win-64
Adding in variants from internal_defaults
Adding in variants from C:\Miniconda\conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from C:\Miniconda\conda_build_config.yaml
INFO:conda_build.metadata:Attempting to finalize metadata for biotite
Attempting to finalize metadata for biotite
Collecting package metadata: ...working... done
Solving environment: ...working... done
Traceback (most recent call last):
  File ".ci_support\build_all.py", line 130, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support\build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support\build_all.py", line 120, in build_folders
    conda_build.api.build(list(d.keys()), config=get_config(arch, channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2314, in build_tree
    notest=notest,
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1324, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 735, in expand_outputs
    for (output_dict, m) in _m.copy().get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "C:\Miniconda\lib\site-packages\conda_build\metadata.py", line 2038, in get_output_metadata_set
    bypass_env_check=bypass_env_check)
  File "C:\Miniconda\lib\site-packages\conda_build\metadata.py", line 721, in finalize_outputs_pass
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 509, in finalize_metadata
    exclude_pattern)
  File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 371, in add_upstream_pins
    permit_unsatisfiable_variants, exclude_pattern)
  File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 359, in _read_upstream_pin_files
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 143, in get_env_dependencies
    channel_urls=tuple(m.config.channel_urls))
  File "C:\Miniconda\lib\site-packages\conda_build\environ.py", line 738, in get_install_actions
    locking=locking, timeout=timeout)
  File "C:\Miniconda\lib\site-packages\conda_build\index.py", line 241, in get_build_index
    platform=subdir)
  File "C:\Miniconda\lib\site-packages\conda\exports.py", line 243, in get_index
    index = _get_index(channel_urls, prepend, platform, use_local, use_cache, unknown, prefix)
  File "C:\Miniconda\lib\site-packages\conda\common\io.py", line 85, in decorated
    return f(*args, **kwds)
  File "C:\Miniconda\lib\site-packages\conda\core\index.py", line 58, in get_index
    index = fetch_index(channel_urls, use_cache=use_cache)
  File "C:\Miniconda\lib\site-packages\conda\core\index.py", line 73, in fetch_index
    sd = SubdirData(Channel(url))
  File "C:\Miniconda\lib\site-packages\conda\core\subdir_data.py", line 60, in __call__
    assert channel.subdir
AssertionError
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Build recipe (maybe vs2008)
```

This is especially peculiar, because the `AppVeyor` build succeeded.
Is the error in my recipe or in the continuous integration script?

File: recipes/biotite/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  number: {{ number }}
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+  skip: True  # [py < 36]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python 3.6
```
Comment (chrisburr) [inline recipes/biotite/meta.yaml]: You shouldn't constrain the python version here as you've stated above that it should be >= 3.6. There will then be two builds for each platform one for Python 3.6 and one for Python 3.7.
```suggestion
    - python
```
This kind of version specifier for `python` should only be used for `noarch: python` packages.

**Comment by padix-key** [comment]
Previously the Python version was not constrained, but then I got the following error:
```
conda.exceptions.UnsatisfiableError: The following specifications were found to be in conflict:
  - numpy=1.13 -> python[version='>=2.7,<2.8.0a0']
  - python=3.7
```
`conda info` shows that there is no Python 3.7 compatible version for NumPy 1.13, that I use for building the package. Hence, I constrained the Python version to 3.6.
Do you see another workaround?
Is there a way to tell the recipe to use the lowest NumPy version compatible with the given Python version?

**Comment by chrisburr** [comment]
You can use something like:
```yaml
    - numpy=1.13  # [py < 37]
    - numpy=1.15  # [py >= 37]
```
(I didn't check which version number is actually appropriate)

**Comment by padix-key** [comment]
Thank you. I incorporated your suggestion.

**Comment by chrisburr** [comment]
There is a [test failure on Windows](https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/22781990#L1919). Can you take a look and fix (or make pytest ignore) it.

**Pull request merged**